src/c-list.h

```diff
@@ -70,7 +70,8 @@ static inline void c_list_init(CList *what) {
  * Return: Pointer to parent container, or NULL.
  */
 #define c_list_entry(_what, _t, _m) \
-        ((_t *)(void *)(((unsigned long)(void *)(_what) ?: \
+        ((_t *)(void *)(((unsigned long)(void *)(_what) ?                       \
+                         (unsigned long)(void *)(_what) :                       \
```

Comment:
This will evaluate `_what` twice, which we really want to avoid if possible. I just pushed a test for this in f1eadf27377ef2f74b3cfd16185f54a219df2aae.